cpp/src/parquet/encoding-benchmark.cc

```diff
@@ -163,4 +172,132 @@ static void BM_DictDecodingInt64_literals(benchmark::State& state) {
 
 BENCHMARK(BM_DictDecodingInt64_literals)->Range(1024, 65536);
 
+std::shared_ptr<::arrow::Array> MakeRandomStringsWithRepeats(size_t num_unique,
```

Comment:
Consider adding a comment here which provides an example of what the output of `MakeRandomStringsWithRepeats` might look like for two example values of `num_unique` and `num_values`,

Reply:
Feel free to disregard this if you don't think it is helpful, but you could consider including `alphabet` (i.e. a list of letters from which to draw randomly) as one of the input arguments to `MakeRandomStringsWithRepeats`, rather than using `::arrow::random_ascii` inside of the function body. This might help facilitate greater flexibility/re-usability of this function in other contexts. For example, there might be a case in the future where a client might want to generate a random string containing Unicode characters, rather than just ASCII.

Reply:
I like this idea and will take this up in [ARROW-4661](https://issues.apache.org/jira/browse/ARROW-4661)